External/CSG/Classes/Node.cs

```diff
@@ -90,44 +89,47 @@ public void Build(List<Polygon> list)
             if (list.Count < 1)
                 return;
 
-            if (this.plane == null || !this.plane.Valid())
+            bool newNode = plane == null || !plane.Valid(); 
+
+            if (newNode)
             {
-                this.plane = new Plane();
-                this.plane.normal = list[0].plane.normal;
-                this.plane.w = list[0].plane.w;
+                plane = new Plane();
+                plane.normal = list[0].plane.normal;
+                plane.w = list[0].plane.w;
             }
 
-
-            if (this.polygons == null)
-                this.polygons = new List<Polygon>();
-
-            List<Polygon> list_front = new List<Polygon>();
-            List<Polygon> list_back = new List<Polygon>();
+            if (polygons == null)
+                polygons = new List<Polygon>();
+                
+            var listFront = new List<Polygon>();
+            var listBack = new List<Polygon>();
 
             for (int i = 0; i < list.Count; i++)
-            {
-                this.plane.SplitPolygon(list[i], this.polygons, this.polygons, list_front, list_back);
-            }
-
-            if (list_front.Count > 0)
-            {
-                if (this.front == null)
-                    this.front = new Node();
+                plane.SplitPolygon(list[i], polygons, polygons, listFront, listBack);
+            
+            
+            if (listFront.Count > 0)
+            {                
+                // SplitPolygon can fail to correctly identify coplanar planes when the epsilon value is too low. When
+                // this happens, the front or back list will be filled and built into a new node recursively. This 
+                // check catches that case and sorts the front/back lists into the coplanar polygons collection.
+                if (newNode && list.SequenceEqual(listFront))
```

Comment:
The `SequenceEqual` here kinda sucks, but in my testing I found that the number new node splits are generally low. Typically less than a 10 or so for each leaf. 

Given that this is an experimental feature and not really intended to be a runtime operation, I didn't spend much time investigating performance impact. In the future it may be worth hashing the list, or better yet, spending some time in `Plane.SplitPolygon` and avoiding this problem in the first place.